test/epochtalk_server_web/controllers/thread_test.exs

```diff
@@ -746,4 +758,179 @@ defmodule Test.EpochtalkServerWeb.Controllers.Thread do
+
+    @tag authenticated: :admin
+    test "given nonexistant thread, does not move thread", %{
+      conn: conn,
+      test_move_board: %{id: board_id}
+    } do
+      response =
+        conn
+        |> post(Routes.thread_path(conn, :move, -1), %{new_board_id: board_id})
+        |> json_response(400)
+
+      assert response["error"] == "Bad Request"
+      assert response["message"] == "Error, cannot move thread"
+    end
+
+    @tag authenticated: :mod
+    test "when authenticated with insufficient priority, given admin thread, throws forbidden read error",
+         %{
+           conn: conn,
+           admin_board_thread: %{post: %{thread_id: thread_id}},
+           test_move_board: %{id: board_id}
+         } do
+      response =
+        conn
+        |> post(Routes.thread_path(conn, :move, thread_id), %{new_board_id: board_id})
+        |> json_response(403)
+
+      assert response["error"] == "Forbidden"
+      assert response["message"] == "Unauthorized, you do not have permission to read"
+    end
+
+    @tag authenticated: :admin
+    test "when authenticated with insufficient priority, given super admin thread, throws forbidden write error",
+         %{
+           conn: conn,
+           super_admin_board_thread: %{post: %{thread_id: thread_id}},
+           test_move_board: %{id: board_id}
+         } do
+      response =
+        conn
+        |> post(Routes.thread_path(conn, :move, thread_id), %{new_board_id: board_id})
+        |> json_response(403)
+
+      assert response["error"] == "Forbidden"
+      assert response["message"] == "Unauthorized, you do not have permission to write"
+    end
+
+    @tag authenticated: :banned
+    test "when authenticated with banned user, throws InvalidPermission forbidden error", %{
+      conn: conn,
+      thread: %{post: %{thread_id: thread_id}},
+      test_move_board: %{id: board_id}
+    } do
+      assert_raise InvalidPermission,
+                   ~r/^Forbidden, invalid permissions to perform this action/,
+                   fn ->
+                     post(conn, Routes.thread_path(conn, :move, thread_id), %{
+                       new_board_id: board_id
+                     })
+                   end
+    end
+
+    @tag authenticated: :mod
+    test "when authenticated with insufficient priority, given admin created thread, throws forbidden error",
+         %{
+           conn: conn,
+           admin_priority_thread: %{post: %{thread_id: thread_id}},
+           test_move_board: %{id: board_id}
+         } do
+      response =
+        conn
+        |> post(Routes.thread_path(conn, :move, thread_id), %{new_board_id: board_id})
+        |> json_response(403)
+
+      assert response["error"] == "Forbidden"
+
+      assert response["message"] ==
+               "Unauthorized, you do not have permission to move another user's thread"
+    end
+
+    @tag :authenticated
+    test "given thread and user who does not moderate the thread, does not move thread", %{
```

Comment:
something more like this would explain the context more:
`when authenticated, without valid permissions, does not move thread`

if the user isn't an input, the route isn't really "given" a user, it's more of a route context